filebeat/harvester/log.go

```diff
@@ -309,16 +312,18 @@ func (h *Harvester) newLogFileReader() (reader.Reader, error) {
 	}
 
 	// Closes reader after timeout or when done channel is closed
+	// This routine is also responsible to properly stop the reader
 	go func() {
 		var closeTimeout <-chan time.Time
 		if h.config.CloseTimeout > 0 {
 			closeTimeout = time.After(h.config.CloseTimeout)
 		}
 
 		select {
-		case <-h.done:
 		case <-closeTimeout:
 			logp.Info("Closing harvester because close_timeout was reached: %s", h.state.Source)
+		case <-h.done:
+		case <-h.harvesterDone:
```

Comment:
wonder if it would make more sense to have this channel local + start go-routine in worker loop itself (Harvest method). e.g. like:

```
harvesterDone := make(chan struct{})
defer close(harvesterDone)

go func() {
   ...
}
```

What's the big difference between `done` and `harvesterDone`? We still need to check `h.done` here then?